packages/core/components/Puck/index.tsx

```diff
@@ -80,38 +84,82 @@ export function Puck<
   }) => ReactElement;
   headerTitle?: string;
   headerPath?: string;
+  viewports?: Viewports;
 }) {
   const historyStore = useHistoryStore();
 
   const [reducer] = useState(() =>
     createReducer<UserConfig>({ config, record: historyStore.record })
   );
 
-  const [initialAppState] = useState<AppState>(() => ({
-    ...defaultAppState,
-    data: initialData,
-    ui: {
-      ...defaultAppState.ui,
-      ...initialUi,
-      // Store categories under componentList on state to allow render functions and plugins to modify
-      componentList: config.categories
-        ? Object.entries(config.categories).reduce(
-            (acc, [categoryName, category]) => {
-              return {
-                ...acc,
-                [categoryName]: {
-                  title: category.title,
-                  components: category.components,
-                  expanded: category.defaultExpanded,
-                  visible: category.visible,
-                },
-              };
-            },
-            {}
-          )
-        : {},
-    },
-  }));
+  const [initialAppState] = useState<AppState>(() => {
+    const initial = { ...defaultAppState.ui, ...initialUi };
+
+    let clientUiState: Partial<AppState["ui"]> = {};
+
+    if (typeof window !== "undefined") {
+      const viewportWidth = window.innerWidth;
+
+      const viewportDifferences = Object.entries(viewports)
+        .map(([key, value]) => ({
+          key,
+          diff: Math.abs(viewportWidth - value.width),
+        }))
+        .sort((a, b) => (a.diff > b.diff ? 1 : -1));
+
+      const closestViewport = viewportDifferences[0].key;
+
+      clientUiState = {
+        // Hide side bars on mobile
+        ...(window.matchMedia("(min-width: 638px)").matches
```

Comment:
Might be worth annotating this value. Assume it's dervied from some calculation of widths of various bit of Puck UI chrome? Possibility of modifying CSS further down the line and not realising this value also needs to change perhaps.

Reply:
Right yes, all of the media query stuff should be updated now that the preview has it's own viewport. One for later.